airflow/ui/src/pages/Dashboard/Metrics.tsx

```diff
@@ -0,0 +1,244 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  Box,
+  Flex,
+  Heading,
+  HStack,
+  VStack,
+  Text,
+  createListCollection,
+  type SelectValueChangeDetails,
+} from "@chakra-ui/react";
+import dayjs from "dayjs";
+import type {
+  TaskInstanceStateCount,
+  DAGRunStates,
+} from "openapi-gen/requests/types.gen";
+import { useState } from "react";
+import { FiCalendar } from "react-icons/fi";
+
+import { useDashboardServiceHistoricalMetrics } from "openapi/queries";
+import { ErrorAlert } from "src/components/ErrorAlert";
+import Time from "src/components/Time";
+import { Select } from "src/components/ui";
+import { capitalize } from "src/utils";
+import { stateColor } from "src/utils/stateColor";
+
+const BAR_WIDTH = 100;
+const BAR_HEIGHT = 5;
+const DAGRUN_STATES: Array<keyof DAGRunStates> = [
+  "queued",
+  "running",
+  "success",
+  "failed",
+];
+const TASK_STATES: Array<keyof TaskInstanceStateCount> = [
+  "queued",
+  "running",
+  "success",
+  "failed",
+  "skipped",
+];
+const timeOptions = createListCollection({
+  items: [
+    { label: "Last 1 hour", value: "1" },
+    { label: "Last 8 hours", value: "8" },
+    { label: "Last 12 hours", value: "12" },
+    { label: "Last 24 hours", value: "24" },
+  ],
+});
+
+type Props = {
+  readonly runs: number;
+  readonly state: string;
+  readonly total: number;
+};
+
+type DagRunStateInfoProps = {
+  readonly dagRunStates: DAGRunStates;
+  readonly total: number;
+};
+
+type TaskRunStateInfoProps = {
+  readonly taskRunStates: TaskInstanceStateCount;
+  readonly total: number;
+};
+
+const StateInfo = ({ runs, state, total }: Props) => {
+  // Calculate the given state as a percentage of total and draw a bar
+  // in state's color with width as state's percentage and remaining width filed as gray
+  const statePercent = total === 0 ? 0 : ((runs / total) * 100).toFixed(2);
+  const stateWidth = total === 0 ? 0 : (runs / total) * BAR_WIDTH;
+  const remainingWidth = BAR_WIDTH - stateWidth;
+
+  return (
+    <VStack align="left" gap={1} mb={4} ml={0} pl={0}>
+      <Flex justify="space-between">
+        <HStack>
+          <Text
+            bg={stateColor[state as keyof typeof stateColor]}
+            borderRadius={15}
+            pb={1}
+            pt={1}
+            px={5}
+          >
+            {runs}
+          </Text>
+          <Text> {capitalize(state)} </Text>
+        </HStack>
+        <Text color="gray.500"> {statePercent}% </Text>
```

Comment:
Let's use the semantic tokens for text colors: https://www.chakra-ui.com/docs/theming/colors#semantic-tokens